make.js

```diff
@@ -0,0 +1,119 @@
+
+require('shelljs/make');
+var fs = require('fs');
+var path = require('path');
+var util = require('./make-util');
+// white list of tasks to make it into the build
+var taskList = require('./make-tasks.json');
+
+// util functions
+var run = util.run;
+var banner = util.banner;
+var rp = util.rp;
+var fail = util.fail;
+var ensureExists = util.ensureExists;
+var pathExists = util.pathExists;
+var buildNodeTask = util.buildNodeTask;
+var addPath = util.addPath;
+var copyTaskResources = util.copyTaskResources;
+var ensureTool = util.ensureTool;
+
+var commonOutputPath = path.join(__dirname, '_common');
+var buildPath = path.join(__dirname, '_build');
+var testPath = path.join(__dirname, '_test');
+
+// add node modules .bin to the path so we can dictate version of tsc etc...
+var binPath = path.join(__dirname, 'node_modules', '.bin');
+if (!test('-d', binPath)) {
+    fail('node modules bin not found.  ensure npm install has been run.');
+}
+addPath(binPath);
```

Comment:
when using `npm test`, the .bin folder gets added to the path automatically. i haven't tested with `npm run ___` but i would guess the same holds true.